controller/scan/harbor.go

```diff
@@ -74,37 +78,90 @@ func (h *harbor) GetAllImages() (map[share.CLUSImage][]string, error) {
 	return images, nil
 }
 
-// TODO: deal with large registries, implement pagination/chunked responses?
 func (h *harbor) getAllRepositories() ([]HarborApiRepository, error) {
-	reqUrl, err := url.JoinPath(h.base.regURL, "api/v2.0/repositories")
+	pageNum := 1
+	pageWhereTotalCountChanged := -1
+	fetchedRepositories := []HarborApiRepository{}
+	totalReposInRegistry := -1
+	for {
+		if pageNum == pageWhereTotalCountChanged {
+			// we've already appended the repos for this page in a previous iteration
+			pageNum++
+			continue
+		}
+		repositories, totalCount, err := h.getPageOfRepositories(pageNum)
```

Comment:
what's returned if a pageNum value bigger than the last page is used to call h.getPageOfRepositories?
Does it return
1. []HarborApiRepository{}, totalCount, {our of page boundary error} , or
2. []HarborApiRepository{}l, totalCount, nil
?